front/src/components/molecules/ChannelAddForm/ChannelAddForm.tsx

```diff
@@ -0,0 +1,125 @@
+    border: '2px solid #B2B1B9',
+    borderRadius: 10,
+  },
+}));
+
+const ChannelAddForm = (): React.ReactElement => {
+  const classes = useStyles();
+  const [passwordInput, setPasswordInput] = useState(false);
+  const [name, setName] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [nameErrorReason, setNameErrorReason] = useState('');
+  const [password, setPassword] = useState('');
+  const [passwordError, setPasswordError] = useState(false);
+  const [passwordErrorReason, setPasswordErrorReason] = useState('');
+  const [passwordConfirm, setPasswordConfirm] = useState('');
+  const [passwordConfirmError, setPasswordConfirmError] = useState(false);
+
+  const setInput = (callback:
+    React.Dispatch<React.SetStateAction<string>>) => (event: React.FormEvent) => {
+    setNameError(false);
+    setPasswordError(false);
+    setPasswordConfirmError(false);
+    callback((event.target as HTMLInputElement).value);
+  };
+
+  const showPasswordForm = () => {
+    setPasswordInput((prevState) => !prevState);
+  };
+
+  const createChannel = () => {
+    if (name.length === 0) {
+      setNameErrorReason('채널명을 입력해주십시오');
+      return setNameError(true);
+    }
+    if (passwordInput && password.length === 0) {
+      setPasswordErrorReason('비밀번호를 입력해주십시오');
+      return setPasswordError(true);
+    }
+    if (passwordInput && password !== passwordConfirm) {
+      return setPasswordConfirmError(true);
+    }
+    // const res = await axios.put({API}/channel);
+    // if (에러 발생) {
+    //   if (body.type === 'name') {
+    //     return setNameErrorReason(res.errorReason);
+    //   }
+    //   if (body.type === 'password') {
+    //     return setPasswordErrorReason(res.errorReason);
+    //   }
+    // }
+    return console.log('create channel');
+  };
+
+  return (
+    <FormControl className={classes.root}>
+      <Grid container justifyContent="center" direction="column" spacing={5}>
+        <Grid item xs={12}>
+          <TextInput
+            name="channel-name"
+            label="채널명"
+            type="text"
+            error={nameError}
+            helperText={nameError ? nameErrorReason : ''}
+            defaultValue={name}
+            onChange={setInput(setName)}
+          />
+        </Grid>
+        <Grid item xs={12}>
+          <Grid container alignItems="center">
+            <CheckBox name="to-private" onChange={showPasswordForm} />
+            <Typo>비공개</Typo>
+          </Grid>
+        </Grid>
+        {
+          passwordInput && (
+            <Grid item xs={12}>
+              <Grid container justifyContent="center" direction="column" spacing={5}>
+                <Grid item xs={12}>
+                  <TextInput
+                    name="channel-password"
+                    label="채널 비밀번호"
+                    type="password"
+                    error={passwordError}
+                    helperText={passwordError ? passwordErrorReason : ''}
```

Comment:
```suggestion
                    error={passwordError !== ''}
                    helperText={passwordError}
```

한 input에 error 처리하는 state를 2개가 아니라 위와 같이 한개로 처리하는게 좋지 않을까요?

Reply:
아하 그게 더 좋겠군요 알겠습니다!